tests/testlib/s2n_connection_test_utils.c

```diff
@@ -180,8 +180,12 @@ int s2n_connections_set_io_pair(struct s2n_connection *client, struct s2n_connec
 
 int s2n_io_pair_close(struct s2n_test_io_pair *io_pair)
 {
-    POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_CLIENT));
-    POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_SERVER));
+    if (fcntl(io_pair->client, F_GETFD) != -1 || errno != EBADF) {
+        POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_CLIENT));
+    }
+    if (fcntl(io_pair->server, F_GETFD) != -1 || errno != EBADF) {
+        POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_SERVER));
+    }
```

Comment:
fcntl seems like overkill, and @goatgoose pointed out offline that the fd might have already been reused. Why not just modify s2n_io_pair_close_one_end to set the fd to -1 after close? Then you just need to check for -1.

Alternatively you could just use the more general pattern of:
```suggestion
    int client_result = s2n_io_pair_close_one_end(io_pair, S2N_CLIENT);
    int server_result = s2n_io_pair_close_one_end(io_pair, S2N_SERVER);
    POSIX_ENSURE(client_result == S2N_SUCCESS, SOME_ERROR_IDK_WHICH);
    POSIX_ENSURE(server_result == S2N_SUCCESS, SOME_ERROR_IDK_WHICH);
```
That will hide the original errors though, so a better solution might be:
```suggestion
    int client_result = s2n_io_pair_close_one_end(io_pair, S2N_CLIENT);
    int client_s2n_errno = s2n_errno;
    int client_errno = errno;

    POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_SERVER));

    s2n_errno = client_s2n_errno;
    errno = client_errno;
    POSIX_GUARD(client_result);
```